### ISSUE TITLE
LaTeX: glossary terms with common definition are rendered with too much vertical whitespace

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#10000]
### Describe the bug

as in title

### How to Reproduce

```
.. glossary::
   :sorted:

   boson
      Particle with integer spin.

   *fermion*
      Particle with half-integer spin.

   tauon
   myon
   electron
      Examples for fermions.

   über
      Gewisse

```

and `make latexpdf`

### Your project

see code snippet

### Screenshots

![Capture d’écran 2021-12-20 à 12 09 48](https://user-images.githubusercontent.com/2589111/146820019-b58a287e-ec41-483a-8013-85e347b221db.png)

### OS

Mac

### Python version

3.8.0

### Sphinx version

4.3.2

**Comment by jfbu** [comment]
do I win a prize for issue #10000? ;-)

**Comment by jfbu** [comment]
Fixed by 9e8e3f9a6efa3f30ecd873a21eca0bfb721588bf (#9999)
